pkg/operator/resources/cluster/apiserver.go

```diff
@@ -212,6 +286,37 @@ func createAPIService(version, namespace string, c client.Client, l logr.Logger)
 	return apiService
 }
 
+func createOrDeletePvcMutatingWebhook(namespace string, c client.Client, l logr.Logger) *admissionregistrationv1.MutatingWebhookConfiguration {
+	log := l.WithName("createOrDeletePvcMutatingWebhook")
+
+	enabled, err := featuregates.IsWebhookPvcRenderingEnabled(c)
+	if err != nil {
+		if !meta.IsNoMatchError(err) && !errors.IsNotFound(err) {
+			log.Error(err, "failed IsWebhookPvcRenderingEnabled")
+		}
+		return nil
+	}
+
+	if enabled {
+		return createPvcMutatingWebhook(namespace, c, l)
+	}
+
+	// If webhook rendering is disabled but the webhook exists, we need to delete it
+	whc := &admissionregistrationv1.MutatingWebhookConfiguration{}
+	crKey := client.ObjectKey{Name: "cdi-api-pvc-mutate"}
+	if err := c.Get(context.TODO(), crKey, whc); err != nil {
+		if client.IgnoreNotFound(err) != nil {
+			log.Error(err, "failed getting")
+		}
+		return nil
+	}
+	if err := c.Delete(context.TODO(), whc); client.IgnoreNotFound(err) != nil {
```

Comment:
`pkg/operator/resources` is really just for static configuration.  Check out [callbacks.go](https://github.com/kubevirt/containerized-data-importer/blob/main/pkg/operator/controller/callbacks.go).  You can write a callback that creates/deletes the webhook based on the feature gate somewhat similar to what we do when creating Routes/SCCs on openshift

Reply:
fixed